### DISCUSSION TITLE
Timestamps

### DISCUSSION THREAD

**Discussion by HValG** [eveningkid/denodb#301] [Q&A]
Hello,

I started using denodb recently, in the getting started part I saw the possibility to use
static timestamps = true;

This should add and update created_at and updated_at field in my table if I'm write.
For that I created a table containing the two fields as:
       created_at: DataTypes.DATE,
       updated_at: DataTypes.DATE

But whenhever I insert something is this table using the api there is null value on both fields.

I am probably doing something wrong but actually I don't know what, do you see something ?

**Comment by eveningkid** [comment]
Hi,

Thank you for asking the question!

Adding:

```js
static timestamps = true;
```

Will automatically create `createdAt` and `updatedAt` fields for you.

You don't have to add them yourself inside your `Model.fields`.

Hope this solves your problem, have a great day :)

**Comment by HValG** [comment]
Hello,

Thank for answer !
Ok I removed the fields, but whenever I create an element and select them all I have:
Wallet { id: 3, name: "test_name", createdat: null, updatedat: null }

Seems like the timestamp is not persisted to db. What is the setup on DB side for this fields to have ?

**Comment by flintforge** [comment]
Hello.
May I advise an option to allow the insertion of the `updatedAt` field only,
when activating timestamps and not both creation and update date at the same time ?
In the case of MongoDB, adding a `createdAt` field is superfluous since it's already encoded in the OID.

**Comment by aimcluster** [comment]
[Research] 2026-03-10 18:20 UTC hourly pass: material update posted. Added actionable acceptance-criteria hardening comments on #26 (completeness + artifact digest + machine-readable reason codes), #5 (timed multi-instance revocation + fixation-rotation evidence), and #9 (workflow identity digest to prevent silent gate drift). No scope change requested; keep #26 -> refreshed #9 provenance bundle -> #8 fail-closed signoff path.
